VirtusizeCore/Sources/Utils/ExpiringCache.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  SessionCache.swift
+//  VirtusizeCore
+//
+//  Copyright (c) 2025-present Virtusize KK
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+private struct CacheEntry {
+	let createdAt: Date
+	let value: Any
+	init(_ value: Any) {
+		self.createdAt = Date()
+		self.value = value
+	}
+}
+
+public class ExpiringCache {
+	public static let shared = ExpiringCache()
+
+	private var cache = [String: CacheEntry]()
```

Comment:
Would there be any thread-safety issue when mutating the cache from multiple functions?

Reply:
That's correct, there would be an issue. This is handled in the following PRs by making `ExpiringCache` an `actor` which automatically handles the concurrency: https://github.com/virtusize/integration_ios/pull/164/files#diff-324867762b7dc54259ab91de0f506170dd2ef4aaec911c1d045d6a718964c3fd